### PR TITLE
docs: add inline docs to public APIs

### PR DESCRIPTION
**Pull request by LayZeeDK**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Public APIs barely have any inline documentation.

Issue Number: N/A

## What is the new behavior?
All software artifacts exposed in public APIs have inline documentation.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

## Other information
TODO:
- [x] `@ngworker/lumberjack`
- [x] `@ngworker/lumberjack/console-driver`
- [x] `@ngworker/lumberjack/http-driver`

### CODE REVIEW

**Comment by LayZeeDK** [comment]
@NachoVazquez I don't think we need to expose `lumberjackHttpDriverConfigToken` in the public API. It's only used internally in the HTTP driver package.

File: libs/ngworker/lumberjack/http-driver/src/lib/configuration/lumberjack-http-driver-internal.config.ts [hunk 1]
```diff
@@ -2,20 +2,24 @@ import { LumberjackLogDriverConfig } from '@ngworker/lumberjack';
 
 import { LumberjackHttpDriverRetryOptions } from './lumberjack-http-driver-retry.options';
 
+/**
+ * Settings used by the HTTP driver.
+ */
```
Comment (NachoVazquez) [inline libs/ngworker/lumberjack/http-driver/src/lib/configuration/lumberjack-http-driver-internal.config.ts:7]: This doc section should also be added to the `lumberjack-http-driver.config`

File: libs/ngworker/lumberjack/src/lib/logging/lumberjack-logger.service.ts [hunk 2]
```diff
@@ -8,34 +8,63 @@ import { LumberjackTimeService } from '../time/lumberjack-time.service';
...
 
+  /**
+   * Create a logger builder for a critical log with the specified message.
+   */
   protected createCriticalLogger(message: string): LumberjackLoggerBuilder<TPayload> {
     return this.createLoggerBuilder(LumberjackLevel.Critical, message);
   }
 
+  /**
+   * Create a logger builder for a critical log with the specified message.
```
Comment (NachoVazquez) [inline libs/ngworker/lumberjack/src/lib/logging/lumberjack-logger.service.ts:31]:
```suggestion
   * Create a logger builder for a debug log with the specified message.
```

**Review by NachoVazquez — CHANGES_REQUESTED** [review]

**Comment by NachoVazquez** [comment]
> @NachoVazquez I don't think we need to expose `lumberjackHttpDriverConfigToken` in the public API. It's only used internally in the HTTP driver package.

Could you include this change in this PR?

**Review by NachoVazquez — APPROVED** [review]

**Pull request merged**